ipykernel/kernelapp.py

```diff
@@ -142,9 +143,10 @@ class IPKernelApp(BaseIPythonApplication, InteractiveShellApp, ConnectionFileMix
     debug_shell_socket = Any()
     stdin_socket = Any()
     iopub_socket = Any()
-    iopub_thread = Any()
-    control_thread = Any()
-    shell_channel_thread = Any()
+
+    iopub_thread: BaseThread
```

Comment:
I think these still need to be defined as traits, e.g. `: BaseThread = Any()` or perhaps even `Instance(BaseThread, allow_none=True)`.

Reply:
I think that will need a `# type: ignore[assign]` as well, or mypy will be unhappy